File: 153-find-minimum-in-rotated-sorted-array/153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0; 
        int right = nums.size() -1;
        int boundary_index = -1; 
        int last = nums[right];
        while (left <= right) {
            
            int mid = (right + left) / 2; 
            
            if(last >= nums[mid]) {
                boundary_index = mid; 
                right = mid - 1;
            }
            else
                left = mid + 1;
        }
        
        
        return nums[boundary_index]; 
    }
};

File: 153-find-minimum-in-rotated-sorted-array/153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        
        boundary_index = -1
        left, right = 0, len(nums) -1
        
        while(left <= right):
            mid = (left + right) // 2
            
            if nums[mid] < nums[-1]:
                right = mid - 1
                boundary_index = mid
            else:
                left = mid + 1
        
        return nums[boundary_index]